client/components/data/query-theme/use-query-single-theme.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'calypso/state';
+import { requestSingleTheme } from 'calypso/state/themes/actions/request-theme';
+import { isRequestingTheme } from 'calypso/state/themes/selectors';
+import { CalypsoDispatch } from 'calypso/state/types';
+import { AppState } from 'calypso/types';
+
+export type SourceType = 'wpcom' | 'wporg' | 'jetpack';
+
+const request =
+	( siteId: number | null, themeId: string, sourceType: SourceType ) =>
+	( dispatch: CalypsoDispatch, getState: AppState ) => {
+		if ( ! isRequestingTheme( getState(), siteId as number, themeId ) ) {
```

Comment:
This is fishy, why are we casting to number if we defined it as nullable? What will happen when the `siteId` is null and casting fails?

Reply:
We already pass the "wpcom" and "wporg" strings as numbers. We should probably update the typings for `isRequestingTheme` in a follow up PR.

Reply:
But I do agree it's sketchy. 😅

Reply:
This is a very frequent situation in Calypso: there are many places where it's guaranteed that the selected `siteId` is present and valid, because the surrounding code guarantees it. For example, when declaring a route:
```js
function showFoo( ctx, next ) {
  const siteId = getSelectedSiteId( context.store.getState() );
  /* ... */
}
page( '/foo/:site', siteSelection, showFoo );
```
Here the `siteSelection` handler makes sure that `showFoo` is called only when the site is validated and loaded and present in Redux state.

But the `getSelectedSiteId` return type is the generic `string | null`, because in general it's possible that no site is selected. The TypeScript type doesn't know the complex context that excludes that possibility for `showFoo`.

The non-null assertion `!` operator could be useful here, but usually it's forbidden by lint configs because it's "not safe".